src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1292,4 +1299,1344 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::add,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract,
+                  EIRPlantLoopHeatPumps::EIRFuelFiredHeatPump::subtract},
+    };
+
+    bool errorsFound = false;
+    std::string &cCurrentModuleObject = state.dataIPShortCut->cCurrentModuleObject;
+    for (auto &classToInput : classesToInput) {
+        cCurrentModuleObject = DataPlant::PlantEquipTypeNames[static_cast<int>(classToInput.thisType)];
+
+        DataLoopNode::ConnectionObjectType objType = static_cast<DataLoopNode::ConnectionObjectType>(
+            getEnumerationValue(BranchNodeConnections::ConnectionObjectTypeNamesUC, UtilityRoutines::MakeUPPERCase(cCurrentModuleObject)));
+
+        auto const instances = state.dataInputProcessing->inputProcessor->epJSON.find(cCurrentModuleObject);
+        if (instances == state.dataInputProcessing->inputProcessor->epJSON.end()) continue;
+        auto &instancesValue = instances.value();
+        for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+            auto const &fields = instance.value();
+            auto const &thisObjectName = instance.key();
+            state.dataInputProcessing->inputProcessor->markObjectAsUsed(cCurrentModuleObject, thisObjectName);
+
+            EIRFuelFiredHeatPump thisPLHP;
+
+            thisPLHP.EIRHPType = classToInput.thisType;
+            std::string companionCoilFieldTag = "companion_heating_heat_pump_name";
+            std::string refCapFieldTag = "nominal_cooling_capacity";
+            if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredHeating) {
+                companionCoilFieldTag = "companion_cooling_heat_pump_name";
+                refCapFieldTag = "nominal_heating_capacity";
+            }
+
+            // A1-A3
+            thisPLHP.name = UtilityRoutines::MakeUPPERCase(thisObjectName);
+            std::string loadSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_inlet_node_name").get<std::string>());
+            std::string loadSideOutletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("water_outlet_node_name").get<std::string>());
+            // Implicit
+            // std::string condenserType = "AIRSOURCE"; // UtilityRoutines::MakeUPPERCase(fields.at("condenser_type").get<std::string>());
+            thisPLHP.waterSource = false;
+            thisPLHP.airSource = false;
+            // A4
+            std::string sourceSideInletNodeName = UtilityRoutines::MakeUPPERCase(fields.at("air_source_node_name").get<std::string>());
+            std::string sourceSideOutletNodeName =
+                "DUMMY_CONDENSER"; // UtilityRoutines::MakeUPPERCase(fields.at("source_side_outlet_node_name").get<std::string>());
+            srand(time(NULL));
+            for (int i = 1; i <= 3; ++i) {
+                sourceSideOutletNodeName += ("_" + std::to_string(rand()));
+            }
```

Comment:
Faster here to just do `std::string sourceSideOutletNodeName = format("DUMMY_CONDENSER_{}_{}_{}", std::to_string(rand()), std::to_string(rand()), std::to_string(rand());`. PS are these many rands necessary?

Reply:
Changed in earlier commits today. The final changes were like this (dropped the `to_string()` when using `format`; and also dropped one `rand()`):
`            std::string sourceSideOutletNodeName = format("DUMMY_CONDENSER_{}_{}", rand(), rand());`